src/components/DatePicker/YearPicker/YearPicker.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react";
+import moment from "moment";
+import times from "lodash/times";
+import { CSSTransition } from "react-transition-group";
+import classnames from "classnames";
+import Button from "../../Button/Button";
+import { YEAR_FORMAT } from "../constants";
+import DateNavigationItemComponent from "../DateNavigationItem/DateNavigationItem";
+import { Moment, Direction } from "../types";
+import styles from "./YearPicker.module.scss";
+
+const transitionOptions = {
+  classNames: "slide-down",
+  timeout: { enter: 400, exit: 400 }
+};
+
+const PAGE_SIZE = 18;
+const BUFFER_FROM_CURRENT_YEAR = 4;
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+interface YearPickerProps {
+  selectedDate?: Moment;
+  isYearBlocked?: (year: number) => boolean;
+  changeCurrentDate: (date: Moment) => void;
+  "data-testid"?: string;
+}
+
+const YearPicker = (props: YearPickerProps) => {
```

Comment:
let's write here the exact props here ({selectedDate, isYearBlocked ...}) like we did in other components for consistency